ext/src/helpers/file_stdin_stream.rs

```diff
@@ -0,0 +1,25 @@
+use std::sync::Arc;
+
+use wasmtime_wasi::{FileInputStream, StdinStream};
+
+pub struct FileStdinStream {
+    file: Arc<File>,
+}
+
+impl FileStdinStream {
+    pub fn new(file: File) -> Self {
+        Self {
+            file: Arc::new(file),
+        }
+    }
+}
+
+impl StdinStream for FileStdinStream {
```

Comment:
The `wasmtime-wasi` API requires types passed to `stdin` to implement `StdinStream` and I couldn't find an existing type that implementing this trait for file input.